v1_17/src/main/java/com/denizenscript/denizen/nms/v1_17/impl/network/handlers/DenizenNetworkManagerImpl.java

```diff
@@ -258,6 +262,33 @@ else if (packet instanceof ClientboundLevelChunkPacket) {
         oldManager.send(packet, genericfuturelistener);
     }
 
+    public boolean processActionbarPacket(Packet<?> packet, GenericFutureListener<? extends Future<? super Void>> genericfuturelistener) {
+        if (!PlayerReceivesActionbarScriptEvent.instance.loaded) {
+            return false;
+        }
+        if (packet instanceof ClientboundSetActionBarTextPacket) {
+            ClientboundSetActionBarTextPacket actionbarPacket = (ClientboundSetActionBarTextPacket) packet;
+            PlayerReceivesActionbarScriptEvent event = PlayerReceivesActionbarScriptEvent.instance;
+            Component baseComponent = actionbarPacket.getText();
+            event.message = new ElementTag(FormattedTextHelper.stringify(Handler.componentToSpigot(baseComponent), ChatColor.WHITE));
+            event.rawJson = new ElementTag(Component.Serializer.toJson(baseComponent));
+            event.system = new ElementTag(false);
+            event.player = PlayerTag.mirrorBukkitPlayer(player.getBukkitEntity());
+            event.modifyMessage = (msg) -> event.message = new ElementTag(msg);
+            event.modifyRawJson = (json) -> event.message = new ElementTag(FormattedTextHelper.stringify(ComponentSerializer.parse(json), ChatColor.WHITE));
+            event.modifyCancellation = (c) -> event.cancelled = c;
+            event.cancelled = false;
+            event.fire();
+            if (!event.cancelled) {
+                Component component = Handler.componentToNMS(FormattedTextHelper.parse(event.message.asString(), ChatColor.WHITE));
+                ClientboundSetActionBarTextPacket newPacket = new ClientboundSetActionBarTextPacket(component);
+                oldManager.send(newPacket, genericfuturelistener);
+            }
+            return true;

```

Comment:
This overrides the packet even if the event doesn't fire and/or doesn't determine anything.
It should only override if there's a modification.